ctrlf.el

```diff
@@ -211,7 +242,10 @@ fails, return nil, but still move point."
                      #'search-backward)
                  (if regexp
                      #'re-search-forward
-                   #'search-forward))))
+                   #'search-forward)))
+         (query (if ctrlf-translate-regexp
+                    (ctrlf--regexp-translate query)
+                  query)))
```

Comment:
Isn't this going to break literal search when translation is enabled?

Reply:
Yes. Now it's fixed.

Reply:
Wait, so fuzzy search isn't enabled when you do a literal search? Isn't that the primary use case? Why would you need to do a fuzzy _regexp_ search, where it's already trivial to put `.*` wherever you want?

Reply:
Let me explain. I almost always use regexp search, even when searching a literal string. Of course if a lot of things need to be escaped I'll switch back to literal search, but for me, it's the default choice. So what I'd like to have for regexp search is I can do fuzzy search when I don't feel want to type much, but at the same time I can easily figure out the translated regexp so I can still do any regexp thing (especially replacing, though I know ctrlf can't do replace by now.).

Doing fuzzy search in literal search is also a nice feature to have, but I want to point out 2 problems:

1. If we do fuzzy search when typing, then we have to use `re-search-forward/backward` under the hood, which seems not the right thing to do. Besides, the replace UX is hard to design, since people usually don't replace when doing fuzzy search, it's hard to imagine what's the reasonable behavior when they do.

2. If we do fuzzy search through another approach, i.e. build the regexp when switching from literal to regexp search, a big drawback is we can't see the result instantly. Also the behavior when switch back to literal search needs to be carefully designed. Sometimes we wanr to keep the original input (like when the user only want to add a word to the query), sometimes we want to keep what we input in regexp search (like when we do regexp first and find it needs too many escapes so switch back to literal search).

But implement fuzzy search only in regexp search feels much better to me: you can do almost the same thing, and you know the mechanism is just some suger on top of ordinary regexp so it feels easier to work with.

Reply:
Sorry, but I think it's too unexpected of a design choice to implement such a drastic change of behavior between two modes in any circumstance. Let's use `re-search-forward` for everything. This simplifies the API and implementation considerably, as there is only one abstraction: a way to transform the user input into a regexp. For literal search, the transformation is `regexp-quote`. For regexp search, it is `identity`. For fuzzy search, there can be another option.

Reply:
I've switched to a more extensible scheme, and it should meet your requirement.